src/persist/src/postgres.rs

```diff
@@ -158,57 +161,73 @@ impl PostgresConsensus {
             },
         );
 
+        let connections_created = config.metrics.connpool_connections_created.clone();
         let pool = Pool::builder(manager)
             .max_size(config.connection_pool_max_size)
-            .post_create(Hook::async_fn(|client, _| {
+            .post_create(Hook::async_fn(move |client, _| {
+                let connections_created = connections_created.clone();
                 Box::pin(async move {
+                    connections_created.inc();
```

Comment:
snuck this in to verify some retry behavior... not really part of this PR but it's a useful metric we don't currently have